Change to Pydantic Version to v2

dasTholo opened the pull request:
change ArangodbClient to python-arango
change python to >py3.10

dump pre-commit version
rename Graph to GrapModel (Name Collision)

dump Shylock Version
add Shylock AsyncDB Class
WIP EdgeModel Funcs like
UPSERT, get_many

change tests to python-arango
fixme 
ENV Variable in pre-commit tests did not run

Commit:
Change Pydantic Version to v2
change ArangodbClient to python-arango
deprecate <py3.10

dump pre-commit version
rename Graph to GrapModel (Name Collision)

dump Shylock Version
add Shylock AsyncDB Class
WIP EdgeModel Funcs like
UPSERT, get_many

change tests to python-arango
fixme ENV Vars in pre-commit tests did not run

dasTholo commented:
I would be happy to receive comments.
This is my first project with Asyncer, so I'm not sure if it will make an improvement.
But aioarangodb is not maintained for a long time and python-arango regularly. Hence my decision to do it this way.

Looking forward to the exchange of thoughts!

dasTholo commented on arangodantic/backends/asyncer_python_arango_backend.py:
this naming isn´t good..
Any suggestions?

bencz replied:
async_arangodb_backend.py is fine

dasTholo committed:
Merge pull request #1 from dasTholo/pydantic2_async

Change Pydantic Version to v2

dasTholo committed:
add upsert func with aql
add KVStoreModel for pure key, value with history

dasTholo committed:
Update pyproject.toml

dasTholo committed:
Update models.py

fix History dependency

dasTholo committed:
Update models.py

dasTholo committed:
Update pyproject.toml

upgrade depends

dasTholo committed:
upgrade depends asyncer

dasTholo committed:
Update models.py

fix insert many EdgeModels

dasTholo committed:
Update project version to 0.4.1 and change 'Project Init and Creating' to 'rye'

Upgrade .pre-commit-config.yaml

TODO change gh workflow to rye

dasTholo committed:
Merge branch 'main' into pydantic2_async

dasTholo committed:
Update project version to 0.4.1 and change 'Project Init and Creating' to 'rye'

- upgrade dependencies
- pin python 3.12
- change from poetry to rye
- Upgrade .pre-commit-config.yaml

dasTholo committed:
Merge remote-tracking branch 'origin/pydantic2_async' into pydantic2_async

dasTholo committed:
Implement ShylockArangoDBBackend for asynchronous ArangoDB operations and update example usage.

dasTholo committed:
fix KeyValueStoreModel to upsert at save

dasTholo committed:
add plain aql execute
upgrade pre-commit

dasTholo committed:
upgrade dependencies

dasTholo committed:
upgrade dependencies

dasTholo committed:
upgrade dependencies

dasTholo committed:
fix CHANGELOG.md
